c10/cuda/CUDAException.cpp

```diff
@@ -38,8 +38,7 @@ void c10_cuda_check_implementation(
         "Device-side assertions were explicitly omitted for this error check; the error probably arose while initializing the DSA handlers.");
   }
 #endif
-
-  TORCH_CHECK(false, check_message);
+  throw c10::DeviceError(err, check_message);
```

Comment:
TORCH_CHECK also provides information about the callsite that this will lose? https://github.com/pytorch/pytorch/blob/e9e1aacef8913bc3baabe502e2858b776e6aea3f/c10/util/Exception.cpp#L109

Reply:
TORCH_CHECK_WITH(DeviceError, false, message)

Reply:
Also, I hope we don't strip error messages on the use side?

Reply:
@ngimel TORCH_CHECK will not preserve the exist code. And callsite was already stripped in the existing code (it was passed as arguments to that function, but ignored) I can add it back as prerequisite/followup PR